### ISSUE TITLE
Version 0.3.0 consumes excessive CPU in RealtimeGaugeDataThread::run()

### ISSUE THREAD

**Issue by bakerkj** [gjr80/weewx-realtime_gauge-data#12]
I recently upgraded to 0.3.0.  I noticed that the CPU usage spiked on my system.  I think, inside of the RealtimeGaugeDataThread::run() method, if there is nothing in either queue the while loop continuously checks each queue over and over again (unlike what happened in 0.2.14 where there was only one queue and it blocked indefinitely).

Here is a potential fix:  Block on the control_queue, but only block on it for a relatively short amount of time (say 1 second).  This way the system will still service the control_queue immediately (like it did in older versions), it will service the result_queue with a 1 second delay.

```
diff --git a/bin/user/rtgd.py b/bin/user/rtgd.py
index 99135a7..7639488 100644
--- a/bin/user/rtgd.py
+++ b/bin/user/rtgd.py
@@ -931,7 +931,7 @@ class RealtimeGaugeDataThread(threading.Thread):
                                 self.forecast_text = _package['payload']
                 # now deal with the control queue
                 try:
-                    _package = self.control_queue.get_nowait()
+                    _package = self.control_queue.get(True, 1.0) # block for one second waiting for package; if none throw Queue.Empty
                 except Queue.Empty:
                     # nothing in the queue so continue
                     pass
```

**Comment by gjr80** [comment]
Thanks for this, was intending to roll it into 0.4.0 which will include a rework of the internal loop packet caching but other things got in the road and this fell by the wayside. 0.4.0 won't be out in the short term so in the interim I have released this change as v0.3.2.

In some basic testing before release I found blocking for 1.0 seconds doubled the time taken to create the `gauge-data.txt` file. Reducing the blocking time to 0.1, 0.01 and 0.001 second reduced the generation time but marginally increased CPU usage (on my VM it went from less than 1% to approx 5%). Not sure why the change in blocking time increases generation time, it shouldn't as far as I can tell. On my VM we are talking of the order of milliseconds so insignificant, not sure how this will equate to real world devices such as RPi1 etc. Will leave this issue open and look at it again as part of 0.4.0
